### ISSUE TITLE
is there any way to use onboard display card for hardware acceleration?

### ISSUE THREAD

**Issue by beyondzvincent** [m1k1o/neko#385]
my setup is intel 13600, no discrete display card,  running ubuntu server, i dont have anything in /dev/dri.
I want to use the onboard display card for hardware acceleration for video, i tried use image: "m1k1o/neko:brave" and ghcr.io/m1k1o/neko/intel-brave, both failed, also tried with or without "NEKO_HWENC: vaapi", either the browser cant open when connect or when i open youtube, it just use pure cpu, not sure if my setting is incorrect, please advise, my docker-compose.yaml is:

`version: "3.4"
services:
  neko:
    image: "m1k1o/neko:brave"
    restart: "unless-stopped"
    shm_size: "2gb"
    volumes:
      - /dev/dri:/dev/dri
    ports:
      - "8080:8080"
      - "52000-52100:52000-52100/udp"
    cap_add:
      - SYS_ADMIN
    environment:
      NEKO_SCREEN: '1920x1080@30'
      NEKO_PASSWORD: neko
      NEKO_PASSWORD_ADMIN: 12345
      NEKO_EPR: 52000-52100
      NEKO_NAT1TO1: 192.168.1.123`

**Comment by bsmithio** [comment]
Do you have anything in /dev/dri on your Docker host? If not you may need to install Intel drivers.

**Comment by beyondzvincent** [comment]
> Do you have anything in /dev/dri on your Docker host? If not you may need to install Intel drivers.
thanks for the advise, i will try that, also is it support hardware acceleration in windows docker? if so, is it also pass the same /dev/dri to docker is ok?

**Comment by yesBad** [comment]
> thanks for the advise, i will try that, also is it support hardware acceleration in windows docker host? if so, is it also pass the same /dev/dri to docker is ok?

Neko has not been tested with Windows and I doubt its the same for Windows. I suggest reading up on that.

**Comment by beyondzvincent** [comment]
> Do you have anything in /dev/dri on your Docker host? If not you may need to install Intel drivers.

Hi, i tested with my other pc that is using  intel 1260p, in this device with ubuntu 22, /dev/dri have card0 and render those, i also can use the igpu for youtube in ubuntu browser, but when i use neko with intel:firefox, it just blank black screen, please advise:

here is the compose file and the log after i connect
[compose.txt](https://github.com/m1k1o/neko/files/14908161/compose.txt)
[nekolog.txt](https://github.com/m1k1o/neko/files/14908162/nekolog.txt)

**Comment by Technetium1** [comment]
@beyondzvincent yes you can do acceleration on windows hosts, but it would **_not_** be a fun time and is not the same exact implementation. No way to specify GPU as far as I know, so you can only use `--gpus=all`. Might only work with Nvidia, but I don't believe that's worth finding out in the first place. It's not worth the development effort to try formally supporting Windows, but please do post details if you can make it work!

**Comment by beyondzvincent** [comment]
> @beyondzvincent yes you can do acceleration on windows hosts, but it would **_not_** be a fun time and is not the same exact implementation. No way to specify GPU as far as I know, so you can only use `--gpus=all`. Might only work with Nvidia, but I don't believe that's worth finding out in the first place. It's not worth the development effort to try formally supporting Windows, but please do post details if you can make it work!

Can you also advise my reply above? about using 1260p with ubuntu, i cannot use hardware acceleration with the integrated graphic

**Comment by Technetium1** [comment]
@beyondzvincent I can't really help, sorry. Try to allocate less memory, maybe some is already taken?
